### ISSUE TITLE
[bug] languageSwitcher not working in examples (sveltekit)

### ISSUE THREAD

**Issue by eikaramba** [opral/monorepo#2216]
### Problem

According to the documentation and also to the repo this should make a simple language switcher:

```js
<script>
	import { availableLanguageTags } from "../paraglide/runtime.js"
	import { i18n } from '$lib/i18n.js'
	import { page } from '$app/stores'
</script>

{#each availableLanguageTags as lang}
	<a 
		href={i18n.route($page.url.pathname)}
		hreflang={lang}
		aria-current={lang === languageTag() ? "page" : undefined}
	>
		{lang}
	</a>
{/each}
```

however the links are all pointing to the same base url for me.

From my understanding it should be  `{i18n.resolveRoute(i18n.route($page.url.pathname),lang)}` no?

### Expected behavior

links should navigate to the language e.g. /de, /en, /fr

### Reproduction

see the official example https://github.com/opral/monorepo/blob/main/inlang/source-code/paraglide/paraglide-js-adapter-sveltekit/example/src/routes/%2Bpage.svelte

### Other information

        "@inlang/paraglide-js": "1.2.5",
        "@inlang/paraglide-js-adapter-sveltekit": "0.3.7",
        "@inlang/plugin-message-format": "2.0.0",

        "@sveltejs/kit": "^2.5.0",
        "svelte": "^4.2.10",
        "vite": "^5.1.1",
        "@sveltejs/vite-plugin-svelte": "^3.0.2",

**Comment by eikaramba** [comment]
but maybe there is actually something not working with the automatic translation system. it works however for every other normal <a> link in my example

**Comment by vytenisstaugaitis** [comment]
Do you have hreflang={lang} in your <a tag?

**Comment by LorisSigrist** [comment]
The code example _is_ correct

> From my understanding it should be {i18n.resolveRoute(i18n.route($page.url.pathname),lang)} no?

Under the hood that is what's happening. The automatic link translations should add the outer `i18n.resolveRoute` automatically so you only need to type `i18n.route($page.url.pathname)`. 

Because it's a language switcher you do need to specify which language you want the link to be in, which you do with the `hreflang` attribute. 

Could you double check & maybe share a repo if the issue persists?

**Comment by eikaramba** [comment]
yeah it's really strange, i am using the code from the top which is 1:1 the one from the docs. i will see in how far i can pinpoint the issue. everything else works and the automatic translation also works in other locations, just not here.

**Comment by LorisSigrist** [comment]
Is the snippet inside some sort of logic block (`{:else}`, `{#each}` etc)? 
There was an issue at one point where that would break the preprocessor, although that should have been fixed.

**Comment by buhodev** [comment]
I had the same issue. languageTag is not imported here `import { availableLanguageTags } from "../paraglide/runtime.js"`, only availableLanguageTags.

**Comment by eikaramba** [comment]
i have the languageTag imported and it is still not translating automatically.
```
<script>
	import { page  } from '$app/stores';
	import { ParaglideJS } from '@inlang/paraglide-js-adapter-sveltekit'
	import { i18n,getLanguageName } from "$lib/utils/i18n"
	import { availableLanguageTags, languageTag,onSetLanguageTag } from "$paraglide/runtime.js"


</script>


<ParaglideJS {i18n}>
	


{#each availableLanguageTags as lang}
	<a 
		href={i18n.route($page.url.pathname)}
		hreflang={lang}
		aria-current={lang === languageTag() ? "page" : undefined}
	>
		{lang}
	</a>
{/each}

</ParaglideJS>
```

this is inside a +layout.svelte file

![grafik](https://github.com/opral/monorepo/assets/1320525/b0449bfd-f4df-4d70-88e5-8ee6279eba8a)

it will take me more time to setup all libraries, config files and see wheather one of them in influencing inlang to make this bug appear(using houdini, unocss,

**Comment by ansbr** [comment]
I have the same trouble, swither is not working at all...
![image](https://github.com/opral/monorepo/assets/58120694/ba0d77b6-7ee8-4a7b-bc22-2b7d9e9831f3)

**Comment by LorisSigrist** [comment]
I'm able to reproduce the issue. (thanks for the repo!) 
I'm still not sure what's causing this, but I'm investigating & will keep you updated

**Comment by feavel1** [comment]
I just created a live preview of the issue. No idea what is going on. https://msng-v5.vercel.app/

**Comment by LorisSigrist** [comment]
Thanks. I'm still puzzled by why this is happening & am still looking.
I've had cases where adding a trailing slash to the links seems to fix the issue, but that's not reliable.

**Comment by LorisSigrist** [comment]
I finally realized what's going on. 

The Link rewrites don't work in the same file as the `<ParaglideJS>` component. The reason is simple: The `<ParaglideJS>` component creates a Svelte Context with everything necessary to translate the links. However, that context can only be read from child components, not the component in which the `<ParaglideJS>` is mounted. 

If you move the Language Switcher into it's own component the issue goes away. 

As for how to fix it, I have a few ideas. I could replace the `<a` elements with my own `<A` component. That way it's guaranteed that the context will be present, as long as the element is inside the `<ParaglideJS>` component.
